pkg/mysql/fields_test.go

```diff
@@ -0,0 +1,167 @@
+	assert.Equal(t, "LONGTEXT", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeLongLong, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "BIGINT", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeMediumBLOB, mysql.Collations[mysql.BinaryCollation])
+	assert.Equal(t, "MEDIUMBLOB", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeMediumBLOB, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "MEDIUMTEXT", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeNewDate, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "DATE", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeNewDecimal, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "DECIMAL", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeNULL, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "NULL", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeSet, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "SET", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeShort, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "SMALLINT", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeString, mysql.Collations[mysql.BinaryCollation])
+	assert.Equal(t, "BINARY", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeString, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "CHAR", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeTime, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "TIME", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeTimestamp, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "TIMESTAMP", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeTiny, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "TINYINT", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeTinyBLOB, mysql.Collations[mysql.BinaryCollation])
+	assert.Equal(t, "TINYBLOB", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeTinyBLOB, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "TINYTEXT", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeVarChar, mysql.Collations[mysql.BinaryCollation])
+	assert.Equal(t, "VARBINARY", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeVarChar, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "VARCHAR", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeVarString, mysql.Collations[mysql.BinaryCollation])
+	assert.Equal(t, "VARBINARY", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeVarString, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "VARCHAR", field.TypeDatabaseName())
+
+	field = createField(mysql.FieldTypeYear, mysql.Collations[mysql.DefaultCollation])
+	assert.Equal(t, "YEAR", field.TypeDatabaseName())
```

Comment:
The above codes have the same format and can be replaced by loop. For example:
```go
	var tests = []struct {
		field     mysql.FieldType
		collation string
		want      string
	}{
		{mysql.FieldTypeBit, mysql.DefaultCollation, "BIT"},
		{mysql.FieldTypeBLOB, mysql.DefaultCollation, "TEXT"},
		{mysql.FieldTypeBLOB, mysql.BinaryCollation, "BLOB"},
		// ...
	}
	for _, tt := range tests {
		field := createField(tt.field, mysql.Collations[tt.collation])
		assert.Equal(t, tt.want, field.TypeDatabaseName())
	}

```

Reply:
> The above codes have the same format and can be replaced by loop. For example:
> 
> ```go
> 	var tests = []struct {
> 		field     mysql.FieldType
> 		collation string
> 		want      string
> 	}{
> 		{mysql.FieldTypeBit, mysql.DefaultCollation, "BIT"},
> 		{mysql.FieldTypeBLOB, mysql.DefaultCollation, "TEXT"},
> 		{mysql.FieldTypeBLOB, mysql.BinaryCollation, "BLOB"},
> 		// ...
> 	}
> 	for _, tt := range tests {
> 		field := createField(tt.field, mysql.Collations[tt.collation])
> 		assert.Equal(t, tt.want, field.TypeDatabaseName())
> 	}
> ```

OK, Thanks.